### PR TITLE
Adding Ollama to Magentic One

### PR DESCRIPTION
**Pull request by MervinPraison**
## Why are these changes needed?

- Added support for a new chat completion client using the Ollama API (OllamaChatCompletionClient).
- Enhanced image handling with base64 encoding for the Ollama API.
- Updated environment-based client creation to accommodate the new Ollama client.

## Related Issue Number

- No specific issue linked. You can mention an issue if applicable.

## Checks

- [x] Documentation updated to reflect new changes.
- [x] Tested locally with Ollama

### CODE REVIEW

**Comment by MervinPraison** [comment]
@microsoft-github-policy-service agree

**Comment by jackgerrits** [comment]
This is great thank you! So it can be used in more places would you mind moving it to `autogen-ext` next to the other model clients?

**Comment by ekzhu** [comment]
Possible to set up Ollama on CI action to test this?

**Comment by MervinPraison** [comment]
@jackgerrits @husseinmozannar @ekzhu Moved the Ollama Client to `autogen-ext` as per the suggestion

File: python/packages/autogen-ext/src/autogen_ext/models/_ollama/_ollama_client.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+class OllamaConfig:
+    base_url: str = "http://localhost:11434"
+    model: str = "llama3.2-vision"
+    temperature: float = 0.7
+    top_p: float = 0.95
+
+class OllamaChatCompletionClient(ChatCompletionClient):
+    def __init__(
+        self,
+        config: OllamaConfig = OllamaConfig(),
```
[the remark this answers is not in the document]
Comment (MervinPraison) [reply python/packages/autogen-ext/src/autogen_ext/models/_ollama/_ollama_client.py]: I could do this in my next commit

File: python/packages/autogen-ext/src/autogen_ext/models/_ollama/_ollama_client.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+)
+
+@dataclass
+class OllamaConfig:
+    base_url: str = "http://localhost:11434"
+    model: str = "llama3.2-vision"
+    temperature: float = 0.7
+    top_p: float = 0.95
+
+class OllamaChatCompletionClient(ChatCompletionClient):
```
[the remark this answers is not in the document]
Comment (MervinPraison) [reply python/packages/autogen-ext/src/autogen_ext/models/_ollama/_ollama_client.py]: I could do this in my next commit

File: python/packages/autogen-ext/src/autogen_ext/models/_ollama/_ollama_client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MervinPraison) [reply python/packages/autogen-ext/src/autogen_ext/models/_ollama/_ollama_client.py]: As a new pull request

**Comment by jackgerrits** [comment]
I am not sure how this pull request got closed by me merging a different PR. I definitely did not mean to do this.

@MervinPraison are you able to create a new PR?

**Comment by MervinPraison** [comment]
Not sure why it happened like that 
No problem 
Sure I will create a new PR

**Comment by MervinPraison** [comment]
@jackgerrits 
Created the pull requested here as per request: https://github.com/microsoft/autogen/pull/4333

**Pull request merged**